port/cpl_aws.cpp

```diff
@@ -1977,6 +2157,17 @@ bool VSIS3HandleHelper::GetConfiguration(
         return true;
     }
 
+    if (!osCredentialProcess.empty())
```

Comment:
this block should likely be moved in the above scope before the return true at line 2157. And it should call GetCredentialsFromProcess() insead of GetOrRefreshTemporaryCredentialsForProcess(), and it should return false if GetCredentialsFromProcess() returns false